File: apps/macos/PasteSheets/Domain/UseCases/SearchUseCase.swift
import Foundation

final class SearchUseCase {

    func search(
        query: String,
        allItems: [PasteItem],
        allDirectories: [DirectoryInfo]
    ) -> (directories: [DirectoryInfo], items: [PasteItem]) {
        let q = query.lowercased()

        let dirs = allDirectories.filter {
            $0.name.lowercased().contains(q)
        }

        let items = allItems.filter {
            $0.content.lowercased().contains(q) ||
            ($0.memo?.lowercased().contains(q) ?? false)
        }

        return (dirs, items)
    }
}

File: apps/windows/PasteSheet/Domain/UseCases/SearchUseCase.cs
using PasteSheet.Domain.Entities;

namespace PasteSheet.Domain.UseCases;

public sealed class SearchUseCase
{
    public (List<DirectoryInfo> Directories, List<PasteItem> Items) Search(
        string query, List<PasteItem> allItems, List<DirectoryInfo> allDirectories)
    {
        var q = query.ToLowerInvariant();

        var dirs = allDirectories
            .Where(d => d.Name.ToLowerInvariant().Contains(q))
            .ToList();

        var items = allItems
            .Where(i => i.Content.ToLowerInvariant().Contains(q)
                        || (i.Memo?.ToLowerInvariant().Contains(q) ?? false))
            .ToList();

        return (dirs, items);
    }
}
